plugins/reach/lib/kReachManager.php

```diff
@@ -49,19 +49,20 @@ private function addingEntryVendorTaskByObjectIds($entryId, $allowedCatalogItemI
 		$catalogItemIdsToAdd = array_unique($allowedCatalogItemIds);
 
 		//If both the entry and reach profile don't exist, there's no need to hit the loop
-		$entry = entryPeer::retrieveByPK($entryId);
-		$entryObjectType = EntryObjectType::ENTRY;
+
+		$entryObjectType = $object instanceof asset ? EntryObjectType::ASSET : EntryObjectType::ENTRY;
```

Comment:
Since this code repeats more than once, lets add a function 
HandlerFactory::getHandlerByObject(object) and take the decision inside, future wise we make have another types - category  / captions / cuepoints....